### ISSUE TITLE
File permissions check don't always seem to work properly - rke2

### ISSUE THREAD

**Issue by rbkaspr** [rancher/cis-operator#125]
Running the rke2-cis-1.5-profile-permissive or -hardened profiles on on rke2 cluster results in somewhat nonsensical results for file permission checks, particularly for cluster certificates. Tests notably affected are CIS 1.1.20 and 4.1.7.

CIS 1.1.20 requires that cluster PKI cert file permissions be 0644 *or more restrictive*. rke2 creates all cert files with either 0644 (literal match) or 0600 (more restrictive). Any files not literally marked 0644 are counted as failures.

CIS 4.1.7 requires that all cluster CA cert permissions be 644 *or more restrictive*. Couple issues here:

- The UI calls out permissions as needing to be 600 or more restrictive, though the test is checking for 644
- Same issue as 1.1.20, where more restrictive permissions are treated as failing
- The test appears to be focused on the permissions of /var/lib/rancher/rke2/server/tls/server-ca.crt. Worker nodes do not have this file (or even the parent tls directory), but are marked as failing unless a file with the appropriate permissions is put in the appropriate place.

**Comment by testn** [comment]
@rbkaspr  I encountered a similar issue. Did you manage to find a workaround?

**Comment by dgiebert** [comment]
I ran into the same problem and did some debugging, this affects 1.1.20 and 1.1.21

`check_files_permissions.sh /var/lib/rancher/rke2/server/tls/*.crt 644` returns false while the filesystem reports

```sh
# ls -al /var/lib/rancher/rke2/server/tls/*.crt
-rw-r--r-- 1 root root 1177 Aug 31 16:56 /var/lib/rancher/rke2/server/tls/client-admin.crt
-rw-r--r-- 1 root root 1186 Aug 31 16:56 /var/lib/rancher/rke2/server/tls/client-auth-proxy.crt
-rw-r--r-- 1 root root  570 Aug 31 16:56 /var/lib/rancher/rke2/server/tls/client-ca.crt
-rw-r--r-- 1 root root 1165 Aug 31 16:56 /var/lib/rancher/rke2/server/tls/client-controller.crt
-rw-r--r-- 1 root root 1181 Aug 31 16:56 /var/lib/rancher/rke2/server/tls/client-kube-apiserver.crt
-rw-r--r-- 1 root root 1149 Aug 31 16:56 /var/lib/rancher/rke2/server/tls/client-kube-proxy.crt
-rw-r--r-- 1 root root 1165 Aug 31 16:56 /var/lib/rancher/rke2/server/tls/client-rke2-cloud-controller.crt
-rw-r--r-- 1 root root 1157 Aug 31 16:56 /var/lib/rancher/rke2/server/tls/client-rke2-controller.crt
-rw-r--r-- 1 root root 1153 Aug 31 16:56 /var/lib/rancher/rke2/server/tls/client-scheduler.crt
-rw-r--r-- 1 root root  595 Aug 31 16:56 /var/lib/rancher/rke2/server/tls/request-header-ca.crt
-rw-r--r-- 1 root root  570 Aug 31 16:56 /var/lib/rancher/rke2/server/tls/server-ca.crt
-rw-r--r-- 1 root root 1404 Aug 31 16:56 /var/lib/rancher/rke2/server/tls/serving-kube-apiserver.crt
```

Seems to me that the `check_file_permissions.sh` needs an eval [here](https://github.com/rancher/security-scan/blob/master/package/helper_scripts/check_files_permissions.sh#L38)

**Comment by rayandas** [comment]
Need security-scan tag and need to update CIS benchmark chart with the new tag

**Comment by vivek-infracloud** [comment]
Tested the issue on rancher 2.7.0 and k8s version v1.24.8+rke2r1.

Test steps:

- Created a rancher server v2.7.0
- Provisioned a 3 node k8s cluster v1.24.8+rke2r1.
- Installed a CIS Benchmark v3.0.1-rc3
- Ran a CIS Profile 
- Tests 1.1.10, 1.1.20, 1.1.21 shows warning.

![Screenshot from 2023-01-04 16-38-15](https://user-images.githubusercontent.com/83208989/210542594-83c31856-53d1-4433-87ce-97b10eb60f25.png)

@dgiebert Should these tests pass? If no, we can close this issue.

**Comment by dgiebert** [comment]
From what I can see, it should be released in version v0.2.10, but I don't know its timeline.

**Comment by rayandas** [comment]
@dgiebert It is going in v0.2.10 only. Right now it’s in pre release https://github.com/rancher/security-scan/tree/v0.2.10-rc1.

**Comment by vivek-infracloud** [comment]
Reopening the issue since the tests above are still failing and those should pass.

**Comment by vivek-infracloud** [comment]
Tested the issue on rancher 2.7-head and k8s version v1.24.9+rke2r2.

Test steps:

- Created a rancher server v2.7.0
- Provisioned a 3 node k8s cluster v1.24.8+rke2r1.
- Installed a CIS Benchmark v3.0.1-rc3
- Ran a CIS Profile 
- Tests 1.1.10, 1.1.20, 1.1.21 are passing.
 
![Screenshot from 2023-01-27 18-04-30](https://user-images.githubusercontent.com/83208989/215088533-f8eccb22-70f6-419e-81e4-5140749238d0.png)
![Screenshot from 2023-01-27 18-03-36](https://user-images.githubusercontent.com/83208989/215088866-85dc9fad-f350-491e-bd5b-36e86d8a68e0.png)

Hence closing the issue.

**Comment by prachidamle** [comment]
@mitulshah-suse @rayandas We should backport the fixes to 2.6

**Comment by rayandas** [comment]
@prachidamle It’s already been backported. Here’s the issue https://github.com/rancher/cis-operator/issues/175
